### DISCUSSION TITLE
RFBridge MQTT Question

### DISCUSSION THREAD

**Discussion by cyclobob** [arendst/Tasmota#11343] [Q&A] [answered]
I have been trying all sorts of variations on this command, but every time I send it via MQTT the bridge responds with command unknown. If I enter it in the console then it works fine.

I first captured the codes from my remote using rfraw 177 then converted the B0 codes to B1 code using the online tool.

And I f I send via the console everything works. I just can't get it to work using MQTT. I am guessing there is some sort of character limitation, but I can't find anything in the notes.

I flashed the bridge with Tasmota 9.3.1 and the RF chip with the latest Portisch software.

This is the command I am trying to use:
cmnd/433bridge/Backlog ‘RfRaw AAB036050809B0065E0154029415CCC092A3A3A3A3B2B2A3B2B2B2A3B2A3A3B2A3B2B2B2B2B2A3A3A3B2A3A3A3A3A3A3A3A3B2B2A3A3B2B2B255; RfRaw 0’

If I enter this into the console:
Backlog ‘RfRaw AAB036050809B0065E0154029415CCC092A3A3A3A3B2B2A3B2B2B2A3B2A3A3B2A3B2B2B2B2B2A3A3A3B2A3A3A3A3A3A3A3A3B2B2A3A3B2B2B255; RfRaw 0’

It works.

Any help appreciated.

Bob

**Comment by sfromis** [comment, marked as answer]
You should have the topic `cmnd/433bridge/Backlog` and then put the commands into the payload.
You should not have those quotes or whatever around the command.

**Reply by cyclobob** [reply]
Thanks for that, but I am not sure how I enter that. If I do 
cmnd/433bridge/Backlog
on its own I get the response 
backlog empty
if I try 
cmnd/433bridge/rfraw AAB036050809B0065E0154029415CCC092A3A3A3A3B2B2A3B2B2B2A3B2A3A3B2A3B2B2B2B2B2A3A3A3B2A3A3A3A3A3A3A3A3B2B2A3A3B2B2B255; RfRaw 0

I get command unknown
Bob

**Reply by cyclobob** [reply]
This also worked perfectly
http://192.168.0.87/cm?cmnd=Backlog RfRaw AAB036050809B0065E0154029415CCC092A3A3A3A3B2B2A3B2B2B2A3B2A3A3B2A3B2B2B2B2B2A3A3A3B2A3A3A3A3A3A3A3A3B2B2A3A3B2B2B255; RfRaw 0

But I am obviously missing something in the MQTT syntax.

Bob

**Reply by sfromis** [reply]
Using MQTT, you must have topic and payload in the **same** message, two fields. Details depend on which tool you use. If a GUI tool, payload is usually in a bigger field below the topic. To use two commands, you **must** have the Backlog command name in the topic, and the commands to be executed split out into the payload, **not** just delimited by a space.

**Reply by cyclobob** [reply]
Got it!

Thanks all working now.

Bob
